modules/loader/src/main/java/com/opengamma/strata/loader/csv/PositionCsvLoader.java

```diff
@@ -378,14 +380,15 @@ private <T extends Position> ValueWithFailures<List<T>> parseFile(CsvIterator cs
               .ifPresent(parsed -> positions.add((T) parsed));
         } else {
           // failed to find the type
-          failures.add(FailureItem.of(
-              FailureReason.PARSING,
-              "CSV position file '{fileName}' contained unknown position type '{type}' at line {lineNumber}",
-              CharSources.extractFileName(charSource),
-              typeRaw,
-              row.lineNumber()));
+          throw new ParseFailureException(FailureReason.UNSUPPORTED, "Unknown '{}', '{}'", POSITION_TYPE_FIELD, typeRaw);
```

Comment:
We need to keep the detail here. While I understand it is in the attributes, we can't rely on that being used, so it must also be in the message.

Reply:
Have ensured all of the existing detail now remains